src/vcpkg/commands.build.cpp

```diff
@@ -1162,17 +1162,15 @@ namespace vcpkg
         std::vector<Path> files;
         std::vector<std::string> hashes;
 
-        size_t i = 0;
-        for (auto& filestr : abi_info.pre_build_info->hash_additional_files)
+        for (size_t i = 0; i < abi_info.pre_build_info->hash_additional_files.size(); ++i)
```

Comment:
`for (size_t i = 0; auto& filestr : abi_info.pre_build_info->hash_additional_files)` would be better but requires c++20 

But I disagree completly with this change. Put an extra block around it if you feel like leaking the variable is such a big issue

Reply:
you're just nitpicking over meaningless things. Doing it one way or another doesn't matter at all. The main point here is clearly replacing `std::vector<string>` with `std::vector<Path>`

Reply:
> you're just nitpicking over meaningless things.

One could say this about this whole PR 🙊 No user would notice a difference in performance but the code is harder to read.

Reply:
It's not at all about perf. IMO a vector of Paths makes a lot more sense when dealing with file paths. Why should callers have to convert it into Path before using it?